File: Chapter-6-Constructor-and-Destructor-Functions/003-a-sample-program-to-introduce-constructors-with-default-arguments.cpp
// a sample program to introduce constructors with default arguments
#include <bits/stdc++.h>

using namespace std;

class myClass
{
    int x;

public:
    myClass(int n = 0) { x = n; }
    int getX() { return x; }
};

int main()
{
    myClass obj(10); // declare with initial value
    myClass obj2;    // delcare without initializer
    cout << "Obj: " << obj.getX() << endl;
    cout << "Obj2: " << obj2.getX() << endl;
    return 0;
}

// output
// Obj: 10
// Obj2 : 0

File: Chapter-6-Constructor-and-Destructor-Functions/003-a-sample-program-to-introduce-constructors-with-default-arguments.py
class MyClass:
    def __init__(self, n=0):
        self.x = n
    
    def get_x(self):
        return self.x

# Creating objects with and without arguments
one = MyClass()
two = MyClass(5)

# Displaying the results
print("one.x = ", one.get_x())
print("two.x = ", two.get_x())